admin/client/src/lib/Injector.js

```diff
@@ -0,0 +1,55 @@
+/**
+ * Using this to map field types to components until we implement dependency injection.
+ */
+class Injector {
+
+  constructor() {
+    // Components registered with the fake DI container.
+    this.components = {};
+  }
+
+  /**
+   * Gets the component matching the passed component name.
+   * Used when a component type is provided bt the form schema.
+   *
+   * @param string componentName - The name of the component to get from the injector.
+   * @return object|null
+   */
+  getComponentByName(componentName) {
+    return this.components[componentName];
+  }
+
+  /**
+   * Default data type to component mappings.
+   * Used as a fallback when no component type is provided in the form schema.
+   *
+   * @param string dataType - The data type provided by the form schema.
+   * @return object|null
+   */
+  getComponentByDataType(dataType) {
+    switch (dataType) {
```

Comment:
This is, well, pretty limited. ;-)

Doesn't block merge, but I would be inclined to use registrations of the form:

```js
injector.register('TextField', TextField);
injector.register('DataType.Text', TextField);
injector.register('HiddenField', HiddenField);
injector.register('DataType.Hidden', HiddenField);
injector.register('GridField', GridField);
injector.register('DataType.Custom', GridField);
```

And then getComponentByDataType can be:

```js
getComponentByDataType(dataType) {
  return this.getComponentByName('DataType.' + dataType);
}
```